src/Data/Semigroup/Foldable.hs

```diff
@@ -36,6 +40,12 @@ import Data.Semigroup hiding (Product, Sum)
 import Data.Semigroup.Foldable.Class
 import Prelude hiding (foldr)
 
+#ifdef MIN_VERSION_generic_deriving
+import Generics.Deriving.Base
+#else
+import GHC.Generics
+#endif
```

Comment:
@RyanGlScott I noticed in a previous CI run that if we don't add the `ifdef`, it fails for GHC 7.0. What do you think, should we drop GHC 7.0, or use `ifdef`?

Reply:
If it's not too much trouble, I think we should consistently use the same CPP in every place where we import `GHC.Generics`.